dvc/tree/osf.py

```diff
@@ -0,0 +1,146 @@
+import logging
+import os
+import threading
+
+from funcy import cached_property, wrap_prop
+
+from dvc.exceptions import DvcException
+from dvc.hash_info import HashInfo
+from dvc.path_info import URLInfo
+from dvc.progress import Tqdm
+from dvc.scheme import Schemes
+
+from .base import BaseTree
+
+logger = logging.getLogger(__name__)
+
+
+class OSFAuthError(DvcException):
+    def __init__(self):
+        message = (
+            "OSF authorization failed. Please check or provide"
+            " osf_username and password."
+            " It is also possible that you do not have access"
+            " to a project."
```

Comment:
This should be enough:
```suggestion
            "OSF authorization failed. Please check or provide"
            " your username and password."
```
Althouth it could also use "See {short link to https://dvc.org/doc/command-reference/remote/modify} for details."